crates/anvil/src/eth/otterscan/api.rs

```diff
@@ -176,7 +180,19 @@ impl EthApi {
         node_info!("ots_getBlockTransactions");
 
         match self.backend.block_by_number_full(number.into()).await? {
-            Some(block) => self.build_ots_block_tx(block, page, page_size).await,
+            Some(block) => {
+                let Block { header, uncles, size, withdrawals, transactions } = block.inner;
+
+                let inner_txs =
+                    transactions.into_transactions().map(|t| t.inner).collect::<Vec<Transaction>>();
+
+                let mut inner_block =
+                    Block { header, uncles, size, withdrawals, ..Default::default() };
+
+                inner_block.transactions = BlockTransactions::from(inner_txs);
+
+                self.build_ots_block_tx(inner_block, page, page_size).await
+            }
```

Comment:
ots types have tx and block generics too, we should update `build_ots_block_tx` instead